File: benchmarks/quicksort/Sort.java
public class Sort {
    static int[] arr;

    static void quicksort(int lo, int hi) {
        if (lo >= hi) return;
        int pivot = arr[hi];
        int i = lo;
        for (int j = lo; j < hi; j++) {
            if (arr[j] <= pivot) {
                int tmp = arr[i]; arr[i] = arr[j]; arr[j] = tmp;
                i++;
            }
        }
        int tmp = arr[i]; arr[i] = arr[hi]; arr[hi] = tmp;
        quicksort(lo, i - 1);
        quicksort(i + 1, hi);
    }

    public static void main(String[] args) {
        int N = 100000;
        arr = new int[N];
        int seed = 42;
        for (int i = 0; i < N; i++) {
            seed = (int)((((long)seed * 1103515245L + 12345L) % 2147483648L));
            arr[i] = seed % 1000000;
        }

        quicksort(0, N - 1);

        int sorted = 1;
        for (int i = 1; i < N; i++) {
            if (arr[i] < arr[i - 1]) sorted = 0;
        }
        System.out.println(sorted);
        System.out.println(arr[0]);
        System.out.println(arr[N - 1]);
    }
}

File: benchmarks/quicksort/sort.py
import sys
sys.setrecursionlimit(200000)

N = 100000
arr = [0] * N
seed = 42
for i in range(N):
    seed = (seed * 1103515245 + 12345) % 2147483648
    arr[i] = seed % 1000000

def quicksort(lo, hi):
    if lo >= hi:
        return
    pivot = arr[hi]
    i = lo
    for j in range(lo, hi):
        if arr[j] <= pivot:
            arr[i], arr[j] = arr[j], arr[i]
            i += 1
    arr[i], arr[hi] = arr[hi], arr[i]
    quicksort(lo, i - 1)
    quicksort(i + 1, hi)

quicksort(0, N - 1)

sorted_flag = 1
for i in range(1, N):
    if arr[i] < arr[i - 1]:
        sorted_flag = 0
print(sorted_flag)
print(arr[0])
print(arr[N - 1])
